### DISCUSSION TITLE
DB2 installation issue with default main.yml settings

### DISCUSSION THREAD

**Discussion by adezanet** [HCL-TECH-SOFTWARE/connections-automation#30] [Q&A]
When installing db2 in the mail.yml file located in connections-automation/roles/third_party/ibm/db2-install/vars/ you reference the file as
 __package_name: "{{ db2_package_name | default('v11.1.4fp5_linuxx64_universal_fixpack.tar.gz') }}" 
 The issue is that on Flexnet when you download connections 7 it will download the file v11.1.4fp5_linuxx64_server_t.tar.gz so the script will fail.
 Also the installation folder is wrong, you reference
__installation_folder: "{{ __extraction_folder }}/universal" 
But it should be: __installation_folder:         "{{ __extraction_folder }}/server_t"

**Comment by dejo1307** [comment]
Thanks a lot for reaching out.

Package name, default one, is for version 11.1.4FP5 for Linux x64 as downloaded from IBMs Fixcentral. However, it can be overwritten by simply setting db2_package_name="v11.1.4fp5_linuxx64_server_t.tar.gz" if that is the name of the package that you are using (or if you are using 11.5 the package name, again, will be different, since we are supporting wide array of versions, not only the single one).

Another issue is the __installation_folder variable. It is following what IBM is releasing, but I am trying to figure out with HCL why is naming in Flexnet different and specifically how is the even the folder structure different. To mitigate this will prepare a PR latest until end of the next week (need to test it properly with different package versions) so that would be rewritable as well.

**Comment by rboccadoro** [comment]
Hi Dejan
thanks a lot for your answer, but when I download the db2 package from the Connections 7 kit on Flexnet I see this
![image](https://user-images.githubusercontent.com/28020376/103899244-6749d180-50f6-11eb-8627-eaa3dda46264.png)
So this is the default name. I don't know where the filename "v11.1.4fp5_linuxx64_universal_fixpack.tar.gz" comes from

**Reply by dejo1307** [reply]
I just got the answer :)

Looks like the names in Flexnet are IBM part numbers. What we used as defaults and for development (and what we are also using internally for our environments) are packages downloaded from Fix Central and looks like they are called differently. I am actually confused myself why and how this package is called this way and on IBMs site another and even internal structure is different. I am just trying to figure it out right now, and will definitely release the PR as soon as we test it. Also, we are working on some elegant way about how those names are mapping, and we started reviewing now (thanks to this) all the names and packages in Flexnet, but that alone will take some time. However, PR should be there in a couple of days and should unblock you. I apologise for eventual issues, I'm also learning on the fly :)

**Reply by rboccadoro** [reply]
No worries :-) As I said is not a show stopper, just need to change the filename in the yaml file. I agree this is not something you want customers to do and  if you can find a way to work around those Flexnet issues would be great ( did I tell you I hate Flexnet ? Try to search for something.... )
If you want me to test your new scripts just let me know, I will be more than happy to help.

**Reply by dejo1307** [reply]
Definitely! :) I just rechecked IBMs site a minute ago, and this is still the name of that specific package we are using here:

fix pack: DB2-linuxx64-universal_fixpack-11.1.4.5-FP005

More Information

DB2 11.1.4 Fix Pack 5 for Linux/x86-64 (64 bit), DB2 Universal Fix Pack

The following files implement this fix.

v11.1.4fp5_linuxx64_universal_fixpack.tar.gz (1.73 GB)

But let me fix it so Flexnet one can be used as well. Will let you know as soon as it is pushed.

**Comment by rboccadoro** [comment]
You are definitely correct, and this inconsistency between IBM site and Flexnet is the root cause, but I guess no one can do anything about this. I believe that all the HCL customers will download the package from Flexnet, though if they are also IBM customers they may get it from the IBM website. You have a hard job to do. I found the same issues with WAS and IIM so if you fix it for DB2 it will hopefully be good also for the other pieces of software.

**Comment by dejo1307** [comment]
So I think the good news are that I've just got confirmed that we are uploading universal packages to Flexnet, so the package from this discussion should be there in a couple of hours. I will update as well as soon as I get it confirmed that it is already uploaded.

**Comment by dejo1307** [comment]
@rboccadoro @adezanet I've got confirmation that it was uploaded, but I don't have access myself to check. Can you confirm please?

**Reply by matteobisi** [reply]
Hi Dejan, yes the universal fixpack for AIX, Win and Linux are there on flexnet

**Reply by rboccadoro** [reply]
Hi Dejan, yes, as Matteo said now the universal fixpack is on Flexnet. I think you should update the documentation to tell explicitly to download that package and not the other one.

**Reply by michallawson** [reply]
Thank you I have found the fix pack on Flexnet, but can not locate the db2jdbc.tar file which the db2 playbook wants to install. Does this have another filename on Flexnet, if so will be grateful if you can tell me what it is, otherwise can you upload it onto flexnet?

Many thanks.

**Reply by rboccadoro** [reply]
That package is not on Flexnet, you need to dowload it from IBM. It is called v11.1.4fp5_jdbc_sqlj.tar.gz
Should I upload it here ?

**Reply by dejo1307** [reply]
Actually, yes, I made the mistake. Let me make the change, I am renaming this package and testing it now (maybe the 'real' one is actually different then the tar file we just made at some point for development and forgot about it).

**Comment by michallawson** [comment]
Dejan,

Thank you and very much appreciated.

Kind regards.

Michael Lawson

**Comment by michallawson** [comment]
Thank you and have downloaded the file from fix central

Kind regards

Michael lawson

**Comment by dejo1307** [comment]
Just quick update here. 

I've fixed it, running the last cycle on my side to test it end to end, PR should be ready in next couple of hours if all is fine with tests.

**Comment by dejo1307** [comment]
Done in https://github.com/HCL-TECH-SOFTWARE/connections-automation/pull/35/files

It is now using proper JDBC package, which should already be uploaded to Flexnet as well.
